node/src/accountant/payment_adjuster/preparatory_analyser/accounts_abstraction.rs

```diff
@@ -0,0 +1,70 @@
+use crate::accountant::payment_adjuster::disqualification_arbiter::DisqualificationArbiter;
+use crate::accountant::payment_adjuster::miscellaneous::data_structures::WeightedPayable;
+use crate::accountant::{AnalyzedPayableAccount, QualifiedPayableAccount};
+
+pub trait DisqualificationAnalysableAccount<Product>: BalanceProvidingAccount
+where
+    Product: BalanceProvidingAccount + DisqualificationLimitProvidingAccount,
+{
+    fn prepare_analyzable_account(
```

Comment:
`provide_account_with_disqualification_limit` is a new name suggestion by @bertllll